openmdao/recorders/sqlite_reader.py

```diff
@@ -501,6 +501,15 @@ def _list_cases_recurse_flat(self, coord=None):
             if case_coord.startswith(coord):
                 cases.append(case_coord)
 
+        if not coord:
+            try:
+                problem_cases = self._problem_cases.list_cases()
+                if problem_cases:
+                    for case in problem_cases:
+                        cases.append(case)
+            except AttributeError:
+                pass
+
```

Comment:
I think this is not the correct fox for this issue.

The bug is that `record_iteration_problem` in `sqlite_recorder.py` does not add an entry to the global_iterations table like the methods for driver, system and solver do.  The `list_cases` function should report every case in the order it was recorded as determined by that table.

Reply:
Fixed